indexer/src/types/transaction.test.ts

```diff
@@ -137,6 +137,167 @@ Deno.test("toTypedEthTx Legacy Transaction", () => {
   assertEquals(ethTx.data, tx.data);
 });
 
+Deno.test("toTypedEthTx Legacy Transaction with v = 28 but not in whitelist (failure case)", () => {
+  // Given
+  const common = new Common({ chain: "mainnet", hardfork: "shanghai" });
+  const tx = new LegacyTransaction(
+    {
+      nonce: 1n,
+      gasPrice: 2n,
+      gasLimit: 3n,
+      to: "0x0000000000000000000000000000000000000001",
+      value: 4n,
+      data: new Uint8Array([0x12, 0x34]),
+    },
+    { common },
+  );
+  const raw = RLP.encode(tx.getMessageToSign());
+
+  const bytesLength = raw.byteLength;
+
+  const starknetTxCalldata: `0x${string}`[] = [
+    "0x1",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    "0x0",
+    `0x${bytesLength.toString(16)}`,
+    ...packCallData(raw),
+  ];
+
+  const starknetTx: Transaction = {
+    invokeV1: {
+      senderAddress: "0x01",
+      calldata: starknetTxCalldata,
+    },
+    meta: {
+      hash: "0x01",
+      maxFee: "0x01",
+      nonce: "0x01",
+      signature: ["0x1", "0x2", "0x3", "0x4", "0x1c"], // 0x1c -> 28
+      version: "1",
+    },
+  };
+
+  // When
+  const ethTx = toTypedEthTx({ transaction: starknetTx }) as LegacyTransaction;
+
+  // Then
+  assertEquals(ethTx, null);
+});
+
+Deno.test("toTypedEthTx Legacy Transaction with v = 28", () => {
```

Comment:
I think that we should have a small condition for v to be 26 or 27 (for pre-eip155) because 28 is impossible for pre-eip155 as per https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md

> If you choose to only hash 6 values, then v continues to be set to {0,1} + 27 as previously.

Reply:
Oh yeah you are right @eugypalu should be 27 or 28, my bad :)

Reply:
Maybe we can still have a check to say that if transaction is pre-eip155 and v is not 27 or 28 we reject it, wdyt?